### PR TITLE
Refine health check

### PR DESCRIPTION
**Pull request by sowu880**
1. Change health check timeout from 60 to 100 seconds.
2. Refine message in FHIR server client to make health checker's log clearer.

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs [hunk 1]
```diff
@@ -214,13 +214,13 @@ private async Task<string> GetResponseFromHttpRequestAsync(Uri uri, string acces
...
-                        _diagnosticLogger.LogError(string.Format("Failed to search from FHIR server: Status code: {0}.", hrEx.StatusCode));
-                        _logger.LogInformation(hrEx, "Failed to search from FHIR server: Status code: {0}.", hrEx.StatusCode);
+                        _diagnosticLogger.LogError(string.Format("Failed to search from FHIR server: Status code: {0}. Reason: {1}", hrEx.StatusCode, hrEx.Message));
+                        _logger.LogInformation(hrEx, "Failed to search from FHIR server: Status code: {0}. Reason: {1}", hrEx.StatusCode, hrEx.Message);
                         break;
                 }
 
                 throw new FhirSearchException(
-                    string.Format(Resource.FhirSearchFailed, uri),
+                    string.Format(Resource.FhirSearchFailed, uri) + $"Reason:{hrEx.Message}",
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs:223]: Just curious, do we need add the exception message to FhirSearchException message as the hrEx is already passed here.

Comment (sowu880) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs:223]: Other components (e.g., health checker) will catch this exception and log exception's message which will ignore inner exception.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs [hunk 2]
```diff
@@ -229,7 +229,7 @@ private async Task<string> GetResponseFromHttpRequestAsync(Uri uri, string acces
                 _logger.LogInformation(bcEx, "Broken circuit while searching from FHIR server. Reason: {0}", bcEx.Message);
 
                 throw new FhirSearchException(
-                    string.Format(Resource.FhirSearchFailed, uri),
+                    string.Format(Resource.FhirSearchFailed, uri) + $"Reason:{bcEx.Message}",
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs:232]: Can we change FhirSearchFailed  string to accept another reason field?

Comment (sowu880) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs]: Updated it.

**Review by moria97 — APPROVED** [review]

**Pull request merged**